.github/workflows/deploy.yml

```
@@ -0,0 +1,43 @@
+name: deploy-book
+
+# Trigger the workflow on push to main branch
+on:
+  push:
+    branches:
+    - master
+    paths:
+    - src/**
+
+# This job installs dependencies, build the book, and pushes it to gh-pages
+jobs:
+  build-and-deploy-book:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [3.8]
+    steps:
+    - uses: actions/checkout@v2
+
+    # Install dependencies
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        pip install -r requirements.txt
+
+    # Build the book
+    - name: Build the book
+      run: |
+        jupyter-book build src
+
+    # Deploy the book's HTML to earth-env-data-science.github.io
+    - name: GitHub Pages action
```

... and then here you can use an if statement to only deploy on master
```suggestion
    - name: GitHub Pages action
      if: github.ref == 'refs/heads/master'
```

copied from.

https://github.com/pangeo-gallery/pangeo-gallery/blob/c038d244006bb772ee2d511391c386311f5c7ec9/.github/workflows/build-and-deploy.yml#L39

---

Done.